File: SignalRDashboard.Web/Scripts/app/models/milliman/GmailStatus.js
(function(signalrdashboardMilliman) {
    signalrdashboardMilliman.GmailStatus = GmailStatus;

    function GmailStatus() {
        // Properties matching data model received via SignalR
        this.lastMail = '';
        this.lastMailTime = '';
        this.mailIsNew = false;

        // Error tracking properties
        this.hasNewMail = false;

        // Error check conditions
        var model = this;
        var errorChecks = [];
        errorChecks.push({ model: model, condition: function() { return model.mailIsNew; }, targetProperty: 'hasNewMail' });

        this.updateFromData = function(data) {            
            signalrdashboard.mapping.map(data, this);
            signalrdashboard.alertHandler.performErrorChecks('GmailStatus', errorChecks);
            //window.setTimeout(function() {
            //    model.hasNewMail = false;
            //    model.mailIsNew = false;
            //}, 20000);
        };
    }
    
})(window.signalrdashboard.milliman || (window.signalrdashboard.milliman = {}));

File: SignalRDashboard.Data.Milliman/Hubs/Models/GmailStatus.cs
using SignalRDashboard.Data.Core.Hubs.Models;

namespace SignalRDashboard.Data.Milliman.Hubs.Models
{
    public class GmailStatus : DashboardHubModel
    {
        private string _lastMail;
        public string LastMail
        {
            get { return _lastMail; }
            set { SetProperty(ref _lastMail, value); }
        }
        private string _lastMailId;
        public string LastMailId
        {
            get { return _lastMailId; }
            set { SetProperty(ref _lastMailId, value); }
        }
        private bool _mailIsNew;
        public bool MailIsNew
        {
            get { return _mailIsNew; }
            set { SetProperty(ref _mailIsNew, value); }
        }
        private string _lastMailTime;
        public string LastMailTime
        {
            get { return _lastMailTime; }
            set { SetProperty(ref _lastMailTime, value); }
        }
    }
}